### PR TITLE
[WIP] Timeout chunk

### PR DESCRIPTION
**Pull request by danielmarbach**
https://github.com/Particular/NServiceBus/pull/3932

Surely you could optimize it a bit more to reduce the empty array allocations for example. But at least a WIP PR

### CODE REVIEW

File: src/NServiceBus.RavenDB/Timeouts/QueryTimeouts.cs [hunk 1]
```diff
@@ -64,23 +64,16 @@ public QueryTimeouts(IDocumentStore documentStore, string endpointName)
                             .Where(t => t.Time >= startSlice)
                             .Where(t => t.Time <= now)
                             .Skip(skipCount)
-                            .Select(t => new
-                            {
-                                t.Id,
-                                t.Time
-                            })
+                            .Select(t => new TimeoutsChunk.Timeout(t.Id, t.Time))
```
Comment (SimonCropp) [inline src/NServiceBus.RavenDB/Timeouts/QueryTimeouts.cs]: @danielmarbach while u r there can u merge the above two where checks

**Comment by mauroservienti** [comment]
Is this a maintainer prio?

**Comment by danielmarbach** [comment]
Of which maintainer group?

**Comment by mauroservienti** [comment]
@Particular/ravendb-persistence-maintainers re this specific PR

**Comment by DavidBoike** [comment]
Rolled up into #261, closing. Thanks @danielmarbach for the head start!

**Pull request closed without merging**